addons/briefing/functions/fnc_addOrbat.sqf

```diff
@@ -1,48 +1,66 @@
-
-_this spawn {
-    uiSleep 10;
-
-    params ["_unit"];
-    TRACE_1("",_unit);
-
-    private _diaryBuilder = [];
-    _diaryBuilder pushBack "<font size='8'>Only accurate at mission start.</font>";
-
-    {
-        if (({isPlayer _x} count (units _x)) > 0) then {
-            if (((side _x) getFriend playerSide) >= 0.6) then {
-                private _color = switch (side _x) do {
-                    case (west): { "#0088EE" }; // use profile colors here
-                    case (east): { "#DD0000" };
-                    case (resistance): { "#00DD00" };
-                    case (civilian): { "#880099" };
-                    default { "#FFFFFF" };
-                };
-                _diaryBuilder pushBack format ["<font color='%1' size='16'>%2</font>", _color, (groupId _x)];
-                {
-                    private _xIcon = getText (configFile >> "CfgVehicles" >> typeOf (vehicle _x) >> "icon");
-                    private _image = getText (configFile >> "CfgVehicleIcons" >> _xIcon);
-                    _diaryBuilder pushBack format ["<img image='%1' width='16' height='16'/><font size='14'>%2</font>", _image, (name _x)];
-                } forEach (units _x);
-            };
-        };
-    } forEach allGroups;
-
-    _unit createDiaryRecord ["diary", ["ORBAT", _diaryBuilder joinString "<br/>"]];
-};
+ #include "script_component.hpp"
+
+if (!hasInterface) exitWith {}; // Exit if not a player.
+
+
+if (!isNil "orbat") then { player removeDiaryRecord ["Diary", orbat] }; // If diary entry already exists, then erase it.
+
+private _allGroupsWithPlayers = [];
+{ if (side _x == side player) then { _allGroupsWithPlayers pushBackUnique group _x } } forEach (call BIS_fnc_listPlayers); // Create array with all player groups matching the player's side
+//{ if (side _x == side player) then { _allGroupsWithPlayers pushBackUnique _x } } forEach allGroups; // DEBUG FOR SINGLEPLAYER
+_allGroupsWithPlayers = [_allGroupsWithPlayers, [], { groupId _x }] call BIS_fnc_sortBy; // Sort array by group callsigns in alphabetical order
+
+// Init variables
+private _str = ""; 
+private _teamColor = '';
+private _roleColor = '';
+
+{
+	private _sideColor = [side  _x] call BIS_fnc_sideColor; // Get RGBA color value of group's side.
+	_sideColor = (_sideColor apply { _x * 2 }) call BIS_fnc_colorRGBAtoHTML; // Brighten the color and convert it to HEX/HTML.
+	_str = _str + format ["<font face='PuristaMedium' size='16' color='%2'>%1</font>", groupId _x, _sideColor] + "<br />"; // Format group callsign with side color.
+
+	{
+		private _role = roleDescription _x;
+		if (_role == "") then { // If roleDescription is set, then truncate. Else use config name.
+			_role = (getText(configFile >> "CfgVehicles" >> (typeOf vehicle _x) >> "displayName"));
+		} else {
+			_role = (_role splitString "@") select 0;
+		};
+
+		if (_x == player) then { _roleColor = '#B21A00' /* red */ } else { _roleColor = '#ffffff' /* white */ }; // Set color of player's name to red
```

Comment:
`player` -> `ACE_PLAYER`? in general all of these requests are because I don't know if we want remote-controlled units to be the side for the ORBAT. If not, ignore